test/unit/test_tpm2_auth_util.c

```diff
@@ -94,27 +94,77 @@ static void test_tpm2_password_util_from_optarg_str_escaped_hex_prefix(void **st
     tpm2_session_close(&session);
 }
 
+FILE mocked_file_stream;
+const char *mocked_file_data = "sekretpasswrd";
+
+FILE * __real_fopen(const char *path, const char *mode);
+FILE * __wrap_fopen(const char *path, const char *mode) {
+    if (strcmp (path, "test_tpm2_auth_util_foobar")) {
+        printf("REAL CALLED\n");
+        return __real_fopen(path, mode);
+    }
+    return mock_ptr_type(FILE*);
```

Comment:
Why not pass a union pointer via will_return() that contains:
```C
union {
bool is_fopen;
struct {
    const char *path;
    const char *mode;
} fopen_args;
struct {
} fememopen_args;
    void *buffer;
    size_t size;
    const char *mode;
};
```

Then all you need to do is mock fopen(), the other calls should stay the same and you have valid data content to check against too if we want to introspect the file or buffer.

Reply:
Disregard this, I am overall fine with your approach, just clean up the other nits.